cou/utils/juju_utils.py

```diff
@@ -649,20 +660,24 @@ async def upgrade_charm(
             switch=switch,
         )
 
-    async def wait_for_active_idle(
+    async def wait_for_idle(
         self,
         timeout: int,
+        status: str = "active",
         idle_period: int = DEFAULT_MODEL_IDLE_PERIOD,
         apps: Optional[list[str]] = None,
         raise_on_blocked: bool = False,
+        raise_on_error: bool = True,
```

Comment:
This PR is way too long. I wish this kind of new functionalities on juju_utils were splitted into different PR